llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -275,12 +309,22 @@ int main(int argc, char **argv) {
   std::vector<std::unique_ptr<Module>> ResultModules;
   std::vector<std::string> ResultSymbolsLists;
 
-  splitModule(*M.get(), GlobalsSet, ResultModules, ResultSymbolsLists);
+  bool NoLists = OutputIRFilesList.empty() && OutputTxtFilesList.empty();
+  bool PerformSplit = !OutputIRFilesList.empty() || NoLists;
+  bool CollectSymbols = !OutputTxtFilesList.empty() || NoLists;
 
   if (BaseOutputFilename == "-")
     BaseOutputFilename = "a.out";
 
-  saveResults(ResultModules, ResultSymbolsLists);
+  if (PerformSplit) {
```

Comment:
please add a comment that the default usage model is that the tool is called twice with the same inputs because clang does not support actions with multiple outputs, hence this workaround with ir/txt generation splitting. It should not bring much extra overhead because ```parseIRFile``` and ```collectKernelsSet``` are small (would be good to estimate) compared to either of
```
    splitModule(*M.get(), GlobalsSet, ResultModules);
    saveResultModules(ResultModules);
```
and 
```
   collectSymbolsLists(GlobalsSet, ResultSymbolsLists);
   saveResultSymbolsLists(ResultSymbolsLists);
```

Reply:
Added.